Automattic/woocommerce-payments
# Respect server-side toggle for platform checkout eligibility

Fixes #3956

#### Changes proposed in this Pull Request

Currently, whether platform checkout can be enabled is determined by an option (`_wcpay_feature_platform_checkout`) set on the merchant site. 1731-gh-Automattic/woocommerce-payments-server, we introduce a server-side flag to control platform checkout eligibility instead. This PR replaces WCPay's local feature flag check with the server-side flag to determine platform checkout eligibility.

#### Testing instructions

This PR depends on and should be tested alongside 1731-gh-Automattic/woocommerce-payments-server in WCPay Server.

- In your WCPay Server database, check the `wcpay_accounts` table. Find the account that corresponds to your merchant site, and take note of its `wcpay_account_id`.
- In your WCPay Server database, go to the `wcpay_account_meta` table and add a row for the account ID you took note of in the previous step. Set the `platform_checkout_eligible` meta key to `0`.
- In your WCPay merchant site, go to WCPay Dev Tools and "clear" your "Account cache contents."
- Look at your "Account cache contents" and verify that the `account`'s `platform_checkout_eligible` field is set to `false`.
- In WCPay settings, note that there's no checkbox to enable platform checkout. On the checkout page, platform checkout should not be available.
- Go back to your `wcpay_account_meta` table and set `platform_checkout_eligible` to `1`.
- In your WCPay merchant site, go to WCPay Dev Tools and "clear" your "Account cache contents."
- Look at your "Account cache contents" and verify that the `account`'s `platform_checkout_eligible` field is set to `true`.
- In WCPay settings, note that there's a checkbox to enable platform checkout. Verify that platform checkout features work only when that checkbox is enabled _and_ `platform_checkout_eligible` is `true`.

Note that, in the WCPay client code, this PR makes the distinction between `platform_checkout_eligible` (the merchant is _allowed_ to enable platform checkout) and `platform_checkout_enabled` (the merchant has _turned on_ platform checkout).

Additional Context: pdpOw2-sg-p2

-------------------


**Post merge**
